### PR TITLE
Gaussian smoothing filter for UniformGrid

### PR DESCRIPTION
**Pull request by imsodin**
Not much to say here: Adds the Gaussian smoothing filter to uniform grids.

For testing I just check that the data is changed at all and shape has not changed.

Also the scalars, preferences filter arguments and correspondingly setting the input of the vtk algo is duplicated so many times, and not always consistently - this should definitely happen in a wrapper function/decorator/...

### CODE REVIEW

File: pyvista/core/filters.py [hunk 1]
```diff
@@ -3367,3 +3367,48 @@ def delaunay_2d(ugrid, tol=1e-05, alpha=0.0, offset=1.0, bound=False):
...
+            Unitless factor to limit the extent of the kernel.
+
+        std_dev : float or iterable, optional
+            Standard deviation of the kernel in pixel units.
+
+        scalars : str, optional
+            Name of scalars to process. Defaults to currently active scalars.
+
+        preference : str, optional
+            When scalars is specified, this is the perfered scalar type to
```
Comment (GuillaumeFavelier) [inline pyvista/core/filters.py]:
```diff
- When scalars is specified, this is the perfered scalar type to
+ When scalars is specified, this is the preferred scalar type to
```

Comment (imsodin) [reply pyvista/core/filters.py]: That typo was copy-pasted (while it could have very well been mine :) ). I fixed it everywhere in filters.py.

**Comment by imsodin** [comment]
Given you have already done it, I'd prefer if you could take care of the example - thanks.

**Comment by banesullivan** [comment]
Great work! Here's a 3D example. In general, this might fit well with the volume rendering examples: https://docs.pyvista.org/examples/02-plot/volume.html

```
import pyvista as pv
from pyvista import examples

data = examples.download_brain()

smoothed_data = data.gaussian_smooth(std_dev=3.)


dargs = dict(clim=smoothed_data.get_data_range(),
             opacity = [0, 0, 0, 0.1, 0.3, 0.6, 1])

n = [100, 150, 200, 245, 255]

p = pv.Plotter(shape=(1, 2), notebook=0)
p.subplot(0, 0)
p.add_text("Original Image", font_size=24)
# p.add_mesh(data.contour(n), **dargs)
p.add_volume(data, **dargs)
p.subplot(0, 1)
p.add_text("Gaussian smoothing", font_size=24)
# p.add_mesh(smoothed_data.contour(n), **dargs)
p.add_volume(smoothed_data, **dargs)
p.link_views()
p.camera_position = [(-162.0, 704.8, 65.02),
                     (90.0, 108.0, 90.0),
                     (0.0068, 0.0447, 0.999)]
p.show()
```

The brain is a little more visible. The opacity mapping could probably be improved

Screen Shot 2019-10-25 at 10 52 43 AM

**Pull request merged**